libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp

```diff
@@ -193,6 +200,15 @@ AP_AdvancedFailsafe::check(uint32_t last_heartbeat_ms, bool geofence_breached, u
     bool gcs_link_ok = ((now - last_heartbeat_ms) < 10000);
     bool gps_lock_ok = ((now - gps.last_fix_time_ms()) < 3000);
 
+    bool terminate_delay_comms = ((now - last_heartbeat_ms) > ((_terminate_delay * 1000) + 10000));
```

Comment:
This is not very intuitive. What you're wanting to do is start a timer after the other timer expires, right? This should really just have it's own timer.